pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -0,0 +1,324 @@
+package buildchain
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	dot "code.google.com/p/gographviz"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+const longDescription = `Output build dependencies of a specific image repository.
+Supported output formats are json, dot, and ast. The default is set to json.
+
+Examples:
+
+    # Build dependency tree for the specified image repository
+    $ openshift ex build-chain [image-repository]:[tag]
+
+    # Build dependency trees for all image repositories
+    $ openshift ex build-chain --all
```

Comment:
Yes - --all is very adminy.  Probably passing no argument should do all in the current namespace, and passing --all should work across all namespaces.

----- Original Message -----
> > +
> > +	buildapi "github.com/openshift/origin/pkg/build/api"
> > +	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
> > +	imageapi "github.com/openshift/origin/pkg/image/api"
> > +)
> > +
> > +const longDescription = `Output build dependencies of a specific image
> > repository.
> > +Supported output formats are json, dot, and ast. The default is set to
> > json.
> > +
> > +Examples:
> > +
> > +    # Build dependency tree for the specified image repository
> > +    $ openshift ex build-chain [image-repository]:[tag]
> > +
> > +    # Build dependency trees for all image repositories
> > +    $ openshift ex build-chain --all
> 
> as it's currently implemented it only does --all for one namespace, should it
> really be doing it for all imagerepos in all namespaces the user has access
> to?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1229/files#r26412826